homeassistant/components/sensor/wunderground.py

```diff
@@ -114,10 +123,15 @@ def __init__(self, rest, condition):
         """Initialize the sensor."""
         self.rest = rest
         self._condition = condition
+        self.entity_id = ENTITY_ID_FORMAT.format("pws_" + self._condition)
 
     @property
     def name(self):
         """Return the name of the sensor."""
+        if self._condition.startswith("forecast"):
+            period = self._condition[-1:]
```

Comment:
Instead of this weird parsing, pass the period as a parameter into the constructor and save it on the instance.